### DISCUSSION TITLE
Machine Info

### DISCUSSION THREAD

**Discussion by RetroGamerFan** [CorsixTH/CorsixTH#2077] [Research Room]
Add machine strength info to Build Rooms selection dialog and the Drug Case Book.

![image](https://user-images.githubusercontent.com/95074358/144209983-13178387-ee28-4456-a8a9-56b6553ec5d3.png)

![image](https://user-images.githubusercontent.com/95074358/144209784-a57d815b-3f73-464c-bed9-d004df542071.png)

Actually, the Drug Case Book could also use a "bar" for medicine effectiveness and show the percentage as well.

**Comment by lewri** [comment]
Converted to an issue, locking

**Reply by lewri** [reply]
Unlocking now again to squash the issue. Discussion is whether the "build" menu should have a strength indicator shown.

**Comment by ARGAMX** [comment]
About Build Rooms selection dialog - Interesting proposal...
But I didn't get what this information on first screen about Strength will give to user?
I don't take decision to build room and where depending on the machine Strength.

But what is really need on Build Rooms screen - is minimum room size.

I don't keep all rooms minimum size in head. So to know that I am going to Build Rooms selection, select the room, and start to planing it... and if I need to compare 2 rooms I have to do it for both. 😁

But the most "funny" part of this when don't have enough money to build this room. You even can't plan it. So you can't to know minimum room size.

In that case you going to Money Manager screen first... 
But sometimes you already borrowed max so you cannot know the know minimum room size! 🤨

I like the idea to add more information. 
But I'd like to know the minimum room size.
We can add and currently machine Strength.
But not sure how this value can be useful.

Second proposal about Case Book looks bit more interesting and useful.

**Comment by ghost** [comment]
Strength is very important if you have Earthquakes enabled. If the strength is too low during an earthquake, you can easily run into destroyed rooms and dead doctors and dead patients.

Displaying the minimum plot area is a good idea. The Rooms are already displayed on a grid background so why not have some semi-transparent blue overlay in one of the corners to indicate the minimum size using the background grid squares.

**Comment by ARGAMX** [comment]
[Quali-T-Gamer](https://github.com/Quali-T-Gamer) I got your idea about Strength. 
Thanks for the explanation. 

So if the problem which you want to solve is to properly maintain hardware and fight against Earthquakes, then I believe we don't need to add Strength on the windows you suggested...

Why?

Because original Theme Hospital already have the flow for that case.

<img width="345" alt="Untitled" src="https://github.com/user-attachments/assets/0b201ac7-0b1a-40d4-a6f7-9701461b60e9">

To check it:

1. Run original game.
2. Press on any machine. You will see red Machine Window.
3. At the right bottom corner of the screen you will the Strength you want to see. But this is not all...
4. Press Ctrl+LeftMouseClick on Machine Name in Machine Window.
5. Press that combination many times and you will iterate through the all machines in the hospital.
6. At the right bottom corner of the screen you will see Strength and Uses Count of the current machine.
7. At Machine Window you will be able to see the bar, which you want to see in Casebook.

I think the main problem that CorsixTH didn't mimic this flow properly:
1. In CorsixTH Ctrl+LeftMouseClick on Machine Name don't work. Only RightMouseClick.
2. There is no right bottom corner indication while iterating.

So from my point of view we don't need to invent new mechanics here.
We just need to properly copy it from original game.

**Comment by ghost** [comment]
The dialog shows the condition of the machine which is different to strength, though a badly maintained machine will likely not survive an Earthquake either.

The info-bar is accurate (and also part of the machine strength changes in the first place).

The confusion raises a new issue. The difference between machine maintenance and machine strength should be made clearer. Strength is improved via research. Later game hospitals, you might not want to buy a room with low strength because of the risk of an Earthquake destroying that room, so you might wait till you've done more research before making a purchase.

Not a massive deal, given Corsixth has the option to pause the game and build/edit/sell rooms.

**Comment by ARGAMX** [comment]
Iterating through machines method #2732
